Subject: [PATCH] fix-run01
---
Index: src/main/java/ru/wordle.txt
===================================================================
diff --git a/src/main/java/ru/wordle.txt b/src/main/java/ru/wordle.txt
deleted file mode 100644
--- a/src/main/java/ru/wordle.txt	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ /dev/null	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
@@ -1,120 +0,0 @@
-apple
-grape
-chair
-table
-water
-dance
-plant
-flame
-sleep
-dream
-stone
-heart
-money
-cloud
-fight
-light
-sweet
-fruit
-music
-beach
-glass
-bloom
-brick
-laugh
-storm
-chess
-queen
-party
-peace
-mango
-block
-shoes
-coins
-stars
-drink
-snake
-piano
-beast
-candy
-piano
-dance
-earth
-flute
-piano
-apple
-honey
-fishy
-beach
-lunar
-stone
-cloud
-grape
-mango
-sweet
-drink
-peach
-money
-laugh
-toast
-chess
-plant
-party
-storm
-heart
-storm
-shelf
-world
-focus
-grape
-charm
-stone
-shoes
-peace
-tiger
-lucky
-jelly
-peach
-shine
-swish
-kicks
-skate
-taste
-flame
-ride
-track
-mount
-field
-showy
-blank
-grain
-plane
-crown
-beast
-funny
-score
-shine
-blank
-zebra
-laser
-waste
-lions
-grasp
-plane
-candy
-piano
-bride
-crest
-whale
-slush
-glove
-bloom
-grape
-beast
-flame
-stone
-grass
-flute
-smile
-cloud
-storm
Index: src/main/resources/simplelogger.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/src/main/resources/simplelogger.properties b/src/main/resources/simplelogger.properties
--- a/src/main/resources/simplelogger.properties	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/resources/simplelogger.properties	(date 1771515297095)
@@ -1,4 +1,4 @@
-org.slf4j.simpleLogger.logFile=wordle-app.log
+org.slf4j.simpleLogger.logFile=./logs/wordle-app.log
 org.slf4j.simpleLogger.defaultLogLevel=info
 org.slf4j.simpleLogger.showDateTime=true
 org.slf4j.simpleLogger.dateTimeFormat=yyyy-MM-dd HH:mm:ss
Index: src/main/resources/logging.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/src/main/resources/logging.properties b/src/main/resources/logging.properties
--- a/src/main/resources/logging.properties	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/resources/logging.properties	(date 1771515229414)
@@ -1,12 +1,12 @@
 handlers=java.util.logging.FileHandler
 .level=INFO
-java.util.logging.FileHandler.level=FINE
+java.util.logging.FileHandler.level=INFO
 java.util.logging.FileHandler.pattern=logs/main.log
 java.util.logging.FileHandler.limit=50000
 java.util.logging.FileHandler.count=1
 java.util.logging.FileHandler.append=true
 java.util.logging.FileHandler.formatter=java.util.logging.SimpleFormatter
 java.util.logging.SimpleFormatter.format=[%1$tT] %3$s %4$s: %5$s%6$s%n
-ru.wordle.level=FINE
+ru.wordle.level=INFO
 ru.wordle.handlers=java.util.logging.FileHandler
 ru.wordle.useParentHandlers=true
Index: src/main/resources/application.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/src/main/resources/application.properties b/src/main/resources/application.properties
--- a/src/main/resources/application.properties	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/resources/application.properties	(date 1771513062441)
@@ -1,8 +1,8 @@
 # Database Configuration
 spring.datasource.driver-class-name=org.postgresql.Driver
-spring.datasource.url=jdbc:postgresql://localhost:5433/wordle?currentSchema=wordle&searchpath=wordle
+spring.datasource.url=jdbc:postgresql://localhost:8432/wordle?currentSchema=wordle
 spring.datasource.username=postgres
-spring.datasource.password=123
+spring.datasource.password=root
 
 # Hibernate Configuration
 spring.jpa.hibernate.ddl-auto=validate
Index: src/main/java/ru/wordle/infrastructure/repository/dao/GameDao.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/infrastructure/repository/dao/GameDao.java b/src/main/java/ru/wordle/infrastructure/repository/dao/GameDao.java
--- a/src/main/java/ru/wordle/infrastructure/repository/dao/GameDao.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/infrastructure/repository/dao/GameDao.java	(date 1771515917273)
@@ -4,14 +4,16 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
 import ru.wordle.infrastructure.entity.GameEntity;
 
 import java.util.Optional;
 import java.util.UUID;
 
+@Repository
 public interface GameDao extends JpaRepository<GameEntity, UUID> {
 
-    @EntityGraph(attributePaths = {"attemptEntities", "attemptEntities.letters"})
+    @EntityGraph(attributePaths = {"attemptEntities"})
     @Query("select g from GameEntity g where g.id = :id")
     Optional<GameEntity> findById(@Param("id") UUID id);
 }
Index: src/main/java/ru/wordle/infrastructure/repository/GameRepositoryJpa.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/infrastructure/repository/GameRepositoryJpa.java b/src/main/java/ru/wordle/infrastructure/repository/GameRepositoryJpa.java
--- a/src/main/java/ru/wordle/infrastructure/repository/GameRepositoryJpa.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/infrastructure/repository/GameRepositoryJpa.java	(date 1771516870387)
@@ -7,13 +7,19 @@
 import org.springframework.transaction.annotation.Transactional;
 import ru.wordle.domain.model.Game;
 import ru.wordle.domain.repository.GameRepository;
+import ru.wordle.infrastructure.entity.AttemptEntity;
 import ru.wordle.infrastructure.entity.GameEntity;
 import ru.wordle.infrastructure.mapper.GameMapper;
+import ru.wordle.infrastructure.repository.dao.AttemptJpaRepository;
 import ru.wordle.infrastructure.repository.dao.GameDao;
 
 import java.time.OffsetDateTime;
+import java.util.HashSet;
+import java.util.Objects;
 import java.util.Optional;
+import java.util.Set;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Repository
 @Profile("jpa")
@@ -22,6 +28,7 @@
 public class GameRepositoryJpa implements GameRepository {
 
     private final GameDao gameDao;
+    private final AttemptJpaRepository attemptJpaRepository;
     private final GameMapper gameMapper;
 
     @Override
@@ -31,6 +38,7 @@
 
         if (entity.getId() == null) {
             entity.setId(UUID.randomUUID());
+            entity.setNew(true);
             entity.setCreatedAt(OffsetDateTime.now());
             game.setGameId(entity.getId().toString());
         }
@@ -50,8 +58,20 @@
 
         try {
             UUID uuid = UUID.fromString(gameId);
-            return gameDao.findById(uuid)
-                    .map(gameMapper::toDomain);
+            GameEntity entity = gameDao.findById(uuid).orElse(null);
+            if (Objects.isNull(entity)) {
+                return Optional.empty();
+            }
+            if (entity.isAttemptsInitialized()) {
+                Set<UUID> attemptIds = new HashSet<>();
+                for (AttemptEntity attemptEntity : entity.getAttemptEntities()) {
+                    attemptIds.add(attemptEntity.getId());
+                }
+
+                attemptJpaRepository.findByIdWithLetters(attemptIds);
+            }
+
+            return Optional.ofNullable(gameMapper.toDomain(entity));
         } catch (IllegalArgumentException e) {
             log.warn("Invalid game ID format: {}", gameId);
             return Optional.empty();
Index: src/main/java/ru/wordle/infrastructure/repository/dao/AttemptJpaRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/infrastructure/repository/dao/AttemptJpaRepository.java b/src/main/java/ru/wordle/infrastructure/repository/dao/AttemptJpaRepository.java
--- a/src/main/java/ru/wordle/infrastructure/repository/dao/AttemptJpaRepository.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/infrastructure/repository/dao/AttemptJpaRepository.java	(date 1771515917279)
@@ -1,10 +1,20 @@
 package ru.wordle.infrastructure.repository.dao;
 
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
 import ru.wordle.infrastructure.entity.AttemptEntity;
 
+import java.util.Collection;
+import java.util.List;
 import java.util.UUID;
 
+@Repository
 public interface AttemptJpaRepository
         extends JpaRepository<AttemptEntity, UUID> {
+
+    @EntityGraph(attributePaths = "letters")
+    @Query("from AttemptEntity where id in :ids")
+    List<AttemptEntity> findByIdWithLetters(Collection<UUID> ids);
 }
Index: src/main/java/ru/wordle/infrastructure/entity/GameEntity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/infrastructure/entity/GameEntity.java b/src/main/java/ru/wordle/infrastructure/entity/GameEntity.java
--- a/src/main/java/ru/wordle/infrastructure/entity/GameEntity.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/infrastructure/entity/GameEntity.java	(date 1771516870392)
@@ -3,13 +3,13 @@
 import javax.persistence.*;
 import lombok.Getter;
 import lombok.Setter;
+import org.springframework.util.CollectionUtils;
 import ru.wordle.domain.model.GameStatus;
 import java.time.OffsetDateTime;
 import java.util.UUID;
 
 import org.hibernate.Hibernate;
 import org.springframework.data.domain.Persistable;
-import javax.persistence.PostLoad;
 import javax.persistence.Transient;
 import java.util.List;
 
@@ -24,7 +24,7 @@
     private UUID id;
 
     @Transient
-    private boolean isNew = true;
+    private boolean isNew;
 
     @Column(name = "secret_word", nullable = false, length = 5)
     private String secretWord;
@@ -43,22 +43,7 @@
     )
     private List<AttemptEntity> attemptEntities;
 
-    @Override
-    public UUID getId() {
-        return id;
-    }
-
-    @Override
-    public boolean isNew() {
-        return isNew;
-    }
-
-    @PostLoad
-    void markNotNew() {
-        this.isNew = false;
-    }
-
     public boolean isAttemptsInitialized() {
-        return Hibernate.isInitialized(attemptEntities);
+        return Hibernate.isInitialized(attemptEntities) && !CollectionUtils.isEmpty(attemptEntities);
     }
 }
Index: src/main/java/ru/wordle/infrastructure/entity/AttemptEntity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/infrastructure/entity/AttemptEntity.java b/src/main/java/ru/wordle/infrastructure/entity/AttemptEntity.java
--- a/src/main/java/ru/wordle/infrastructure/entity/AttemptEntity.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/infrastructure/entity/AttemptEntity.java	(date 1771516870375)
@@ -2,14 +2,20 @@
 
 import lombok.Getter;
 import lombok.Setter;
-import lombok.Getter;
-import lombok.Setter;
-import org.springframework.data.domain.Persistable;
 import org.hibernate.Hibernate;
-
+import org.springframework.data.domain.Persistable;
+import org.springframework.util.CollectionUtils;
 
-
-import javax.persistence.*;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.Transient;
 import java.util.List;
 import java.util.UUID;
 
@@ -46,26 +52,10 @@
     private List<LetterEntity> letters;
 
     @Transient
-    private boolean isNew = true;
-
-    @Override
-    public UUID getId() {
-        return id;
-    }
-
-    @Override
-    public boolean isNew() {
-        return isNew;
-    }
-
-    @PostLoad
-    @PostPersist
-    private void markNotNew() {
-        this.isNew = false;
-    }
+    private boolean isNew;
 
     public boolean isLettersInitialized() {
-        return Hibernate.isInitialized(letters);
+        return Hibernate.isInitialized(letters) && !CollectionUtils.isEmpty(letters);
     }
 
 }
\ No newline at end of file
Index: src/main/java/ru/wordle/api/error/GlobalExceptionHandler.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/api/error/GlobalExceptionHandler.java b/src/main/java/ru/wordle/api/error/GlobalExceptionHandler.java
--- a/src/main/java/ru/wordle/api/error/GlobalExceptionHandler.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/api/error/GlobalExceptionHandler.java	(date 1771515100489)
@@ -1,5 +1,6 @@
 package ru.wordle.api.error;
 
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.HttpStatus;
 import org.springframework.validation.FieldError;
 import org.springframework.web.bind.MethodArgumentNotValidException;
@@ -17,14 +18,14 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@Slf4j
 @RestControllerAdvice
 public class GlobalExceptionHandler {
 
-    private static final Logger log = LoggerFactory.getLogger(GlobalExceptionHandler.class);
-
     @ExceptionHandler(MethodArgumentNotValidException.class)
     @ResponseStatus(HttpStatus.BAD_REQUEST)
     public ErrorDto handleValidation(MethodArgumentNotValidException ex) {
+        log.error(ex.getMessage(), ex);
         Map<String, String> fields = ex.getBindingResult().getFieldErrors().stream()
                 .collect(Collectors.toMap(FieldError::getField, FieldError::getDefaultMessage, (a, b) -> a, LinkedHashMap::new));
 
@@ -34,18 +35,21 @@
     @ExceptionHandler(InvalidWordException.class)
     @ResponseStatus(HttpStatus.BAD_REQUEST)
     public ErrorDto handleInvalidWordException(InvalidWordException ex) {
+        log.error(ex.getMessage(), ex);
         return new ErrorDto("invalid_word", ex.getMessage(), null);
     }
 
     @ExceptionHandler(WordNotInDictionaryException.class)
     @ResponseStatus(HttpStatus.UNPROCESSABLE_ENTITY)
     public ErrorDto handleWordNotInDictionaryException(WordNotInDictionaryException ex) {
+        log.error(ex.getMessage(), ex);
         return new ErrorDto("not_in_dictionary", ex.getMessage(), null);
     }
 
     @ExceptionHandler(GameAlreadyFinishedException.class)
     @ResponseStatus(HttpStatus.BAD_REQUEST)
     public ErrorDto handleGameAlreadyFinished(GameAlreadyFinishedException ex) {
+        log.error(ex.getMessage(), ex);
         return new ErrorDto("game_already_finished", ex.getMessage(), null);
     }
 
@@ -53,13 +57,14 @@
     @ExceptionHandler(UserNotStartedGameException.class)
     @ResponseStatus(HttpStatus.CONFLICT)
     public ErrorDto handleUserNotStarted(UserNotStartedGameException ex) {
+        log.error(ex.getMessage(), ex);
         return new ErrorDto("game_not_started", ex.getMessage(), null);
     }
 
     @ExceptionHandler(Exception.class)
     @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
     public ErrorDto handleOther(Exception ex) {
-        log.error("Unexpected error in API", ex);
+        log.error(ex.getMessage(), ex);
         return new ErrorDto("internal_error", "unexpected error", null);
     }
 }
Index: src/main/java/ru/wordle/infrastructure/repository/AttemptRepositoryJpa.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/infrastructure/repository/AttemptRepositoryJpa.java b/src/main/java/ru/wordle/infrastructure/repository/AttemptRepositoryJpa.java
--- a/src/main/java/ru/wordle/infrastructure/repository/AttemptRepositoryJpa.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/infrastructure/repository/AttemptRepositoryJpa.java	(date 1771516870382)
@@ -12,6 +12,7 @@
 import ru.wordle.infrastructure.repository.dao.AttemptJpaRepository;
 
 import java.util.ArrayList;
+import java.util.Objects;
 import java.util.UUID;
 
 @Repository
@@ -26,7 +27,13 @@
     public void save(Attempt attempt) {
 
         AttemptEntity attemptEntity = new AttemptEntity();
-        attemptEntity.setId(UUID.randomUUID());
+        if (Objects.isNull(attempt.getId())) {
+            attemptEntity.setId(UUID.randomUUID());
+            attemptEntity.setNew(true);
+            attempt.setId(attemptEntity.getId().toString());
+        } else {
+            attemptEntity.setId(UUID.fromString(attempt.getId()));
+        }
         attemptEntity.setGameId(UUID.fromString(attempt.getGameId()));
         attemptEntity.setAttemptNumber(attempt.getAttemptNumber());
 
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
--- a/pom.xml	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/pom.xml	(date 1771515402963)
@@ -67,6 +67,11 @@
             <version>${jackson.version}</version>
         </dependency>
         <dependency>
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-jsr310</artifactId>
+            <version>${jackson.version}</version>
+        </dependency>
+        <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-core</artifactId>
             <version>${jackson.version}</version>
Index: src/main/java/ru/wordle/infrastructure/configuration/DatabaseConfig.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/infrastructure/configuration/DatabaseConfig.java b/src/main/java/ru/wordle/infrastructure/configuration/DatabaseConfig.java
--- a/src/main/java/ru/wordle/infrastructure/configuration/DatabaseConfig.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/infrastructure/configuration/DatabaseConfig.java	(date 1771514477216)
@@ -4,9 +4,12 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
 import javax.sql.DataSource;
 
 @Configuration
+@EnableTransactionManagement
 public class DatabaseConfig {
 
     @Value("${spring.datasource.url}")
Index: src/main/java/ru/wordle/infrastructure/configuration/JpaConfig.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/infrastructure/configuration/JpaConfig.java b/src/main/java/ru/wordle/infrastructure/configuration/JpaConfig.java
--- a/src/main/java/ru/wordle/infrastructure/configuration/JpaConfig.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/infrastructure/configuration/JpaConfig.java	(date 1771514825449)
@@ -1,5 +1,6 @@
 package ru.wordle.infrastructure.configuration;
 
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
@@ -19,53 +20,27 @@
 
 @Configuration
 @Profile("jpa")
-@EnableTransactionManagement
-@EnableJpaRepositories(basePackages = "ru.wordle.infrastructure.repository.dao")
-
+@EnableJpaRepositories("ru.wordle")
 public class JpaConfig {
 
-    @Value("${jpa.packages-to-scan}")
-    private String packagesToScan;
-
-    @Value("${hibernate.dialect}")
-    private String dialect;
-
-    @Value("${hibernate.hbm2ddl.auto}")
-    private String hbm2ddlAuto;
-
-    @Value("${hibernate.show_sql}")
-    private String showSql;
-
-    @Value("${hibernate.format_sql}")
-    private String formatSql;
-
-    @Value("${hibernate.jdbc.lob.non_contextual_creation}")
-    private String nonContextualCreation;
-
     @Bean
-    @DependsOn("liquibase")
     public LocalContainerEntityManagerFactoryBean entityManagerFactory(DataSource dataSource) {
         LocalContainerEntityManagerFactoryBean em = new LocalContainerEntityManagerFactoryBean();
         em.setDataSource(dataSource);
-        em.setPackagesToScan(packagesToScan);
-
+        em.setPackagesToScan("ru.wordle");
         HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
+        vendorAdapter.setShowSql(true);
         em.setJpaVendorAdapter(vendorAdapter);
-
-        Properties props = new Properties();
-        props.put("hibernate.dialect", dialect);
-        props.put("hibernate.hbm2ddl.auto", hbm2ddlAuto);
-        props.put("hibernate.show_sql", showSql);
-        props.put("hibernate.format_sql", formatSql);
-        props.put("hibernate.jdbc.lob.non_contextual_creation", nonContextualCreation);
-
-        em.setJpaProperties(props);
         return em;
     }
 
     @Bean
-    public PlatformTransactionManager transactionManager(EntityManagerFactory emf) {
-        return new JpaTransactionManager(emf);
+    public PlatformTransactionManager transactionManager(DataSource dataSource, EntityManagerFactory emf) {
+        JpaTransactionManager transactionManager = new JpaTransactionManager();
+        transactionManager.setDataSource(dataSource);
+        transactionManager.setEntityManagerFactory(emf);
+        return transactionManager;
     }
+
 
 }
Index: src/main/resources/db/changelog/008-add-fk-letters-attempts.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/db/changelog/008-add-fk-letters-attempts.xml b/src/main/resources/db/changelog/008-add-fk-letters-attempts.xml
new file mode 100644
--- /dev/null	(date 1771516457776)
+++ b/src/main/resources/db/changelog/008-add-fk-letters-attempts.xml	(date 1771516457776)
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="
+        http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.25.xsd">
+
+    <changeSet id="008-add-position-column-to-letters" author="vkosovanov">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <columnExists tableName="letters" columnName="position"/>
+            </not>
+        </preConditions>
+
+        <addColumn tableName="letters">
+            <column name="position" type="int" remarks="Номер буквы">
+                <constraints nullable="false"/>
+            </column>
+        </addColumn>
+    </changeSet>
+</databaseChangeLog>
Index: src/main/resources/db/changelog/db.changelog-master.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/db/changelog/db.changelog-master.xml b/src/main/resources/db/changelog/db.changelog-master.xml
--- a/src/main/resources/db/changelog/db.changelog-master.xml	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/resources/db/changelog/db.changelog-master.xml	(date 1771516568065)
@@ -5,18 +5,13 @@
         xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
                             http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.25.xsd">
 
-<include file="001-create-words-table.xml" relativeToChangelogFile="true"/>
-
+    <include file="001-create-words-table.xml" relativeToChangelogFile="true"/>
     <include file="003-create-games-table.xml" relativeToChangelogFile="true"/>
-
     <include file="002-load-words-data.xml" relativeToChangelogFile="true"/>
-
     <include file="004-create-attempts-table.xml" relativeToChangelogFile="true"/>
-
     <include file="005-create-letters-table.xml" relativeToChangelogFile="true"/>
-
     <include file="006-add-fk-attempts-game.xml" relativeToChangelogFile="true"/>
-
     <include file="007-add-fk-letters-attempts.xml" relativeToChangelogFile="true"/>
+    <include file="008-add-fk-letters-attempts.xml" relativeToChangelogFile="true"/>
 
 </databaseChangeLog>
Index: src/main/resources/db/changelog/006-add-fk-attempts-game.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/db/changelog/006-add-fk-attempts-game.xml b/src/main/resources/db/changelog/006-add-fk-attempts-game.xml
--- a/src/main/resources/db/changelog/006-add-fk-attempts-game.xml	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/resources/db/changelog/006-add-fk-attempts-game.xml	(date 1771513610826)
@@ -7,7 +7,7 @@
         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.25.xsd">
 
     <changeSet id="006-add-fk-attempts-game" author="Ivan">
-
+        <dropForeignKeyConstraint baseTableName="attempts" constraintName="fk_attempts_game"/>
         <addForeignKeyConstraint
                 baseTableName="attempts"
                 baseColumnNames="game_id"
Index: src/main/java/ru/wordle/domain/model/Game.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/domain/model/Game.java b/src/main/java/ru/wordle/domain/model/Game.java
--- a/src/main/java/ru/wordle/domain/model/Game.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/domain/model/Game.java	(date 1771516680153)
@@ -56,6 +56,7 @@
         List<Letter> letters = new ArrayList<>();
         for (int i = 0; i < secretWord.length(); i++) {
             Letter letter = new Letter();
+            letter.setPosition(i);
             char charIsSecretWord = secretWord.charAt(i);
             char charIsEnterWord = enterWord.charAt(i);
 
Index: src/main/resources/db/changelog/007-add-fk-letters-attempts.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/db/changelog/007-add-fk-letters-attempts.xml b/src/main/resources/db/changelog/007-add-fk-letters-attempts.xml
--- a/src/main/resources/db/changelog/007-add-fk-letters-attempts.xml	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/resources/db/changelog/007-add-fk-letters-attempts.xml	(date 1771513610817)
@@ -7,7 +7,7 @@
         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.25.xsd">
 
     <changeSet id="007-add-fk-letters-attempts" author="Ivan">
-
+        <dropForeignKeyConstraint baseTableName="letters" constraintName="fk_letters_attempt"/>
         <addForeignKeyConstraint
                 baseTableName="letters"
                 baseColumnNames="attempt_id"
Index: src/main/java/ru/wordle/domain/model/Letter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/domain/model/Letter.java b/src/main/java/ru/wordle/domain/model/Letter.java
--- a/src/main/java/ru/wordle/domain/model/Letter.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/domain/model/Letter.java	(date 1771516680161)
@@ -1,37 +1,18 @@
 package ru.wordle.domain.model;
 
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@ToString
 public class Letter {
 
     private Character value;
+
     private LetterStatus status;
-    private Integer position;  // Добавлено поле для позиции
-
-    public Character getValue() {
-        return value;
-    }
 
-    public void setValue(Character value) {
-        this.value = value;
-    }
+    private Integer position;
 
-    public LetterStatus getStatus() {
-        return status;
-    }
-
-    public void setStatus(LetterStatus status) {
-        this.status = status;
-    }
-
-    public Integer getPosition() {
-        return position;
-    }
-
-    public void setPosition(Integer position) {
-        this.position = position;
-    }
-
-    @Override
-    public String toString() {
-        return "{" + value + " = " + status + "}";
-    }
 }
\ No newline at end of file
Index: src/main/java/ru/wordle/infrastructure/repository/dao/WordDao.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/ru/wordle/infrastructure/repository/dao/WordDao.java b/src/main/java/ru/wordle/infrastructure/repository/dao/WordDao.java
--- a/src/main/java/ru/wordle/infrastructure/repository/dao/WordDao.java	(revision dea3e4bd8d9825847fb3ad2eb5e87e046f7d162f)
+++ b/src/main/java/ru/wordle/infrastructure/repository/dao/WordDao.java	(date 1771514367057)
@@ -6,6 +6,7 @@
 
 import java.util.UUID;
 
+@Repository
 public interface WordDao extends JpaRepository<WordEntity, UUID> {
 
     long countByWord(String word);
